src/components/Analysis/Purposes/LoadedPurposesList.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useMemo } from 'react'
+import cx from 'classnames'
+
+import List from 'cozy-ui/transpiled/react/MuiCozyTheme/List'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
+
+import {
+  computeAggregatedTimeseries,
+  sortTimeseriesByCO2GroupedByPurpose,
+  computeCO2Timeseries
+} from 'src/lib/timeseries'
+import { formatCO2 } from 'src/lib/trips'
+import { makeChartProps } from 'src/components/Analysis/helpers'
+import AnalysisListItem from 'src/components/Analysis/AnalysisListItem'
+import PieChart from 'src/components/PieChart/PieChart'
+
+const LoadedPurposesList = ({ timeseries }) => {
+  const { t } = useI18n()
+  const { isMobile } = useBreakpoints()
+
+  const aggregatedTimeseries = useMemo(
+    () => computeAggregatedTimeseries(timeseries),
+    [timeseries]
```

Comment:
let's use useCallback here and inside the 4 functions of the file

Reply:
J'ai pas compris... tu veux remplacer les 4 useMemo par useCallback ?

Reply:
```
  const aggregatedTimeseries = useCallback(computeAggregatedTimeseries(timeseries), [timeseries])
  ```

https://medium.com/@jan.hesters/usecallback-vs-usememo-c23ad1dc60

> Side note: Now - knowing about first class functions - you should understand why useCallback(fn, deps) is equivalent to useMemo(() => fn, deps).

Reply:
top merci